app/controllers/uploads_controller.rb

```diff
@@ -1,56 +1,80 @@
-      file_type = params[:file_type]
-      program = params[:program]
-      year = params[:year]
-
-      @upload = Upload.new(artifact: artf, file_type: file_type, 
-                           program: program, year: year)
-      @upload.save!(validate: false)
-      # TODO: rename errors on the Upload class, so we can remove this validate: false stuff
-
-      FileProcessJob.perform_later(@upload.id.to_s)
-
-      redirect_to upload_path(@upload)
-    ensure
-      File.delete(params[:file].tempfile)
-      # rails 4 activejob adapter is not fully implemented
-      #  wrt the run at a specific time later with sucker_punch
-      # RecordCleanupJob.set(wait: 10.minutes).perform_later
-      RecordCleanupJob.perform_in(600) # run the cleanup job in 10 mins (600 sec)
-    end
+    artf = Artifact.new(file: params[:file])
+    file_type = params[:file_type]
+    program = params[:program]
+    year = params[:year]
+
+    @upload = Upload.new(artifact: artf, file_type: file_type,
+                         program: program, year: year)
+    @upload.save!(validate: false)
+    # TODO: rename errors on the Upload class, so we can remove this validate: false stuff
+
+    FileProcessJob.perform_later(@upload.id.to_s)
+
+    redirect_to upload_path(@upload)
+  ensure
+    File.delete(params[:file].tempfile)
+    # rails 4 activejob adapter is not fully implemented
+    #  wrt the run at a specific time later with sucker_punch
+    # RecordCleanupJob.set(wait: 10.minutes).perform_later
+    RecordCleanupJob.perform_in(600) # run the cleanup job in 10 mins (600 sec)
   end
 
   def show
     @upload = Upload.find(params[:id])
 
-    redirect_to(root_path) and return unless @upload
+    redirect_to(root_path) && return unless @upload
     return unless @upload.completed?
 
-    measure_ids = @upload.qrda_files.collect{ |file| file.get_measure_ids }.flatten.uniq
-
     @bundle = BUNDLES[@upload.year]
-    @measures = @bundle ? @bundle.measures.in(hqmf_id: measure_ids) : []
 
     @files = @upload.qrda_files
   end
+
+  def should_switch_highlight(data_key, pop_key, specifics, rationale)
+    # check final specifics approach
+    if specifics && specifics[pop_key]
+      pop_final_specifics = []
+      unless specifics[pop_key].empty?
+        # first specfics entry for this population is final specifics array
+        pop_final_specifics = specifics[pop_key][0]
+      end
+      return check_specifics(data_key, rationale, pop_final_specifics)
+    end
+    false
+  end
+
+  def check_specifics(data_key, rationale, pop_final_specifics)
+    if rationale[data_key][:specifics] &&
+       !rationale[data_key][:specifics].empty?
+      # get this data criteria's specific info
+      dc_specifics = rationale[data_key][:specifics][0]
+      dc_specifics.each do |dc_spec|
+        # find any data criteria specific entry that is not included
+        # in the population final specifics
+        true if (dc_spec != '*') && !pop_final_specifics.include?(dc_spec)
```

Comment:
Eep- thanks for the catch! Looks like I did some overzealous code reduction on the last pass!